pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1436,51 +1446,92 @@ func (c *Cloud) InstanceType(ctx context.Context, nodeName types.NodeName) (stri
 }
 
 // getCandidateZonesForDynamicVolume retrieves  a list of all the zones in which nodes are running
-// It currently involves querying all instances
 func (c *Cloud) getCandidateZonesForDynamicVolume() (sets.String, error) {
 	// We don't currently cache this; it is currently used only in volume
 	// creation which is expected to be a comparatively rare occurrence.
 
 	// TODO: Caching / expose v1.Nodes to the cloud provider?
 	// TODO: We could also query for subnets, I think
+	zones, err := c.getCachedCandidateZonesForDynamicVolume()
+	if err != nil {
```

Comment:
getCandidateZonesForDynamicVolume looks bit weird in terms of structure because the logic is split in two:
- a one-line call to getCachedCandidateZonesForDynamicVolume
- if it fails a similar logic in the function itself

Wouldn't it be better to also use a function for the error case?

Reply:
Done!